spring-integration-core/src/main/java/org/springframework/integration/support/MutableMessageHeaders.java

```diff
@@ -0,0 +1,61 @@
+package org.springframework.integration.support;
+
+import org.springframework.messaging.MessageHeaders;
+
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * A MessageHeaders that permits direct access to and modification of the
+ * header map.
+ *
+ * @author Stuart Williams
+ * @since 4.2
+ */
+class MutableMessageHeaders extends MessageHeaders {
+
+    private static final long serialVersionUID = 3084692953798643018L;
+
+    /**
+     * @param headers map
+     */
+    public MutableMessageHeaders(Map<String, Object> headers) {
+        super(headers);
+    }
+
+    /**
+     * @param headers map
+     * @param id of message
+     * @param timestamp of message
+     */
+    public MutableMessageHeaders(Map<String, Object> headers, UUID id, Long timestamp) {
+        super(headers, id, timestamp);
+    }
+
+    @Override
+    public Map<String, Object> getRawHeaders() {
+        return super.getRawHeaders();
+    }
```

Comment:
@pidster This override is not needed (after Artem's polish which changed it back to protected). Or did you intend to provide public access to the raw headers? Not sure that's needed since they have full access via the `entrySet()` etc., do you have a specific use case in mind?

Also tabs not spaces (one line in MM too `this.payload = payload`)